### DISCUSSION TITLE
Sharing of Code

### DISCUSSION THREAD

**Discussion by christopherpeers** [cyberbotics/wrestling#74] [General]
Hello,

In regard to this rule - A team may perfectly have only one single member. However, an individual cannot be a member of several teams in the leaderboard. Any team suspected to break this rule will be removed.

However, the rule doesn't mention giving code to other people who aren't part of your team. 

Does this mean that code sharing is allowed?

**Comment by omichel** [comment]
I believe that sharing code publicly should be allowed.
However, sharing code privately (e.g, with some teams only, but not all) seems a little bit unfair indeed.
But, it may be difficult to prove that some code was shared privately.
Nevertheless, I would agree to add some extra rule saying something like:
"Sharing code privately with another team is not allowed. Any team suspected to break this rule will be removed."
And:
"Sharing code publicly is allowed as long as it is immediately announced on this [GitHub Discussions](https://github.com/cyberbotics/wrestling/discussions)."
Do you happen to suspect any team to do this?

**Reply by christopherpeers** [reply]
Action #4573 was fun to watch, but we would like to hear your opinion.

**Reply by omichel** [reply]
So, could @Andres2311 @Roboticaparaeducar from [Topotoropo](https://github.com/Roboticaparaeducar) and @danielebg @AndresVargasA from [KAIZEN](https://github.com/danielebg) tell us if they are sharing some code?

**Reply by JustLuckyRobotics** [reply]
i find it even more suspicious that topotoropo deleted their participant.py in one commit to then commit moments later before the first action even begun. the second commit was an entire new participant.py with a smaller size, all happening shortly after you posted this message pinging them. obviously I'm only going by the info in the action data, but to me this sounds extremely suspect of hiding the incriminating evidence without even engaging in this thread.

Extremely undesirable situation but unless they engage in this thread, reviewing of the commit history/changes would be the only way to know for sure.

**Reply by AndresVargasA** [reply]
hello, we were looking at the rules and as @christopherpeers says there is none that does not allow sharing code between teams, we did it with the intention of testing certain features as the computers of each team do not have enough performance to download the world and test, the only way we have is when we do push and see the fight, but nevertheless, we understand the dissatisfaction of other teams and we will not share code between us again.

**Reply by JustLuckyRobotics** [reply]
So you two are two teams working together to the point you are sharing code. you are effectively 1 team with 2 accounts.

By your logic so long as i don't give my friend our actual code, i can explain to them how to do things with pseudocode. no actual code though.

**Reply by christopherpeers** [reply]
@omichel, where is the line drawn concerning teams co-operating together before they are actually just one team occupying more than one slot of the top 32 teams?

**Reply by omichel** [reply]
I agree there is not yet any explicit rule preventing from passing code from one team to another. However, the rules clearly state that a participant cannot be part of two different teams and what you did sounds like one participant from a team is also helping another team, which is somehow breaking the spirit of this rule and is unfair to other competitors.

@AndresVargasA, @danielebg, @Andres2311 @Roboticaparaeducar: what do you propose to handle this case? I see two possibilities:
1. You promise you won't do it again. And if you happen to do it it again, I will ban the two teams, without any warning.
2. I remove one of the two teams from the leaderboard (please tell me which one) and you merge your two teams together.

Feel free to propose any alternative solution you may find fair.

Please let me know what you decide. Without any news from you by the end of this week, I will make my own decision.

**Reply by AndresVargasA** [reply]
Cordial greetings,
due to so many disagreements with the other teams we have taken the decision to eliminate one of the two teams and unite, the team that we will eliminate will be topotoropo, this in order to avoid further inconvenience to the other teams, and we apologize for any kind of problem caused.

**Reply by omichel** [reply]
Thank you for your quick reaction. There is no problem. I will remove topotoropo from the list of participant.
